src/bundle/Core/Command/ExpireUserPasswordsCommand.php

```diff
@@ -0,0 +1,383 @@
+                . 'fix the issue and rerun this command</error>'
+            );
+
+            throw $e;
+        }
+
+        $output->writeln(sprintf(
+            '<info>Expired passwords of %d user(s)</info>',
+            $processedUsersCount
+        ));
+
+        if ($force) {
+            $this->repository->commit();
+        } else {
+            $this->repository->rollback();
+
+            $output->writeln(
+                '<info>No changes made. If you want to proceed rerun '
+                . 'this command with --force flag.</info>'
+            );
+        }
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * @param array<int> $userIds
+     * @param array<int> $userGroupIds
+     * @param array<string> $userContentTypeIdentifiers
+     *
+     * @return array<\eZ\Publish\API\Repository\Values\Content\Query\Criterion>
+     */
+    private function getCriteria(
+        array $userIds,
+        array $userGroupIds,
+        array $userContentTypeIdentifiers
+    ): array {
+        $criteria = [];
+
+        if (!empty($userIds)) {
+            $criteria[] = new Query\Criterion\ContentId($userIds);
+        }
+
+        if (!empty($userGroupIds)) {
+            $criteria[] = new Query\Criterion\ParentLocationId($userGroupIds);
+        }
+
+        if (!empty($userContentTypeIdentifiers)) {
+            $criteria[] = new Query\Criterion\ContentTypeIdentifier($userContentTypeIdentifiers);
+        }
+
+        return $criteria;
+    }
+
+    /**
+     * @param array<int> $userIds
+     * @param array<int> $userGroupIds
+     * @param array<string> $userContentTypeIdentifiers
+     */
+    private function summarizeSearchCriteria(
+        OutputInterface $output,
+        $userIds,
+        $userGroupIds,
+        $userContentTypeIdentifiers
+    ): void {
+        $output->writeln('<info>Criteria used to find users:</info>');
+
+        if (!empty($userIds)) {
+            $output->writeln(
+                sprintf("<info>\tUser ID: %s</info>", implode(', ', $userIds))
+            );
+        }
+
+        if (!empty($userGroupIds)) {
+            $output->writeln(
+                sprintf("<info>\tUser Group ID: %s</info>", implode(', ', $userGroupIds))
+            );
+        }
+
+        if (!empty($userContentTypeIdentifiers)) {
+            $output->writeln(
+                sprintf(
+                    "<info>\tUser Content Type Identifier: %s</info>",
+                    implode(', ', $userContentTypeIdentifiers)
+                )
+            );
+        }
+    }
+
+    /**
+     * @param array<string, mixed> $fieldSettings
+     * @param array<string, mixed> $validatorConfiguration
+     */
+    private function updateContentType(
+        ContentType $contentType,
+        array $fieldSettings,
+        array $validatorConfiguration,
+        FieldDefinition $fieldDefinition
+    ): void {
+        $this->repository->sudo(
+            static function (
+                Repository $repository
```

Comment:
the same remark for `sudo` with callback here as well. We want to avoid using Repository getters as much as possible.